sdks/python/apache_beam/yaml/yaml_provider.py

```diff
@@ -231,9 +231,17 @@ def provider_from_spec(cls, spec):
           result.to_json = lambda: spec
         return result
       except Exception as exn:
-        raise ValueError(
-            f'Unable to instantiate provider of type {type} '
-            f'at line {SafeLineLoader.get_line(spec)}: {exn}') from exn
+        if isinstance(exn, ModuleNotFoundError) and config.get('requires_gcp',
```

Comment:
Why is this needed in addition to the import guards in yaml_enrichment.py?

It would also be better to get an error on an attempted use rather than a print statement whenever this is imported without [gcp] installed.

Reply:
Ah you're right this isn't needed anymore. This was my initial solution for another test failure before I realized the imports needed to be disabled entirely to fix another test suite. I think this is redundant and misleading as you said at this point.

@reeba212 I added some comments on what to remove!